dm_control/locomotion/walkers/assets/dog_v2/add_muscles.py

```diff
@@ -0,0 +1,435 @@
+
+        site_name = mtu + "_" + str(counter)
+        # Create the site
+        closest_geom_body.add("site", name=site_name,
+                              pos=array_to_string(site_matrix[:3, 3]),
+                              group='4', dclass='connector')
+
+        counter += 1
+        spatial.add('site', site=site_name)
+
+        prev_point = point
+        prev_body = closest_geom_body
+
+        # check for wrapping geoms
+        if idx != len(paths) - 1:
+          geoms = physics.named.data.geom_xpos.axes.row.names
+          dist = 100000000000
+          closest_g = None
+          closest_geom_pos = None
+          for g_name in geoms:
+            if 'wrapping' in g_name:
+              pos = physics.named.data.geom_xpos[g_name]
+              new_dist = np.linalg.norm(pos - point)
+              if (mtu in lateral or mtu in neck) and \
+                  g_name not in wrap_geoms_legs and \
+                  pos[2] < point[2] and \
+                  new_dist < dist:
+                closest_g = g_name
+                dist = new_dist
+                closest_geom_pos = pos
+              elif (mtu in flexors_back or
+                    mtu in flexors_front or
+                    mtu in extensors_front or
+                    mtu in extensors_back) and \
+                  pos[2] < point[2] and \
+                  dist > 0.01 and \
+                  new_dist < dist and \
+                  g_name in wrap_geoms_legs:
+                # in the legs we are interested only in wrapping
+                # geometries that are lower than the site we
+                # are adding
+                closest_g = g_name
+                dist = new_dist
+                closest_geom_pos = pos
+
+          if dist < 0.1:
+            if (mtu in flexors_back or mtu in flexors_front) \
+                  and closest_geom_pos[2] < point[2]:
+              spatial.add('geom', geom=closest_g,
+                          sidesite=closest_g + '_backward')
+            elif (mtu in extensors_back or mtu in extensors_front) \
+                    and closest_geom_pos[2] < point[2]:
+              spatial.add('geom', geom=closest_g,
+                          sidesite=closest_g + '_forward')
+            elif mtu in neck:
+              pos1 = physics.named.data.site_xpos[closest_g + '_forward']
+              d1 = np.linalg.norm(pos1 - point)
+              pos2 = physics.named.data.site_xpos[closest_g + '_backward']
+              d2 = np.linalg.norm(pos2 - point)
+              if d1 < d2:
+                spatial.add('geom', geom=closest_g,
+                            sidesite=closest_g + '_forward')
+              else:
+                spatial.add('geom', geom=closest_g,
+                            sidesite=closest_g + '_backward')
+            elif mtu in torso and dist < 0.05:
+              spatial.add('geom', geom=closest_g,
+                          sidesite=closest_g + '_forward')
+
+    if muscle_dynamics in ['Millard', 'Sigmoid']:
```

Comment:
nit: consider creating an enum class for muscle dynamics to make your code more readable and extensible.